generate-third-party-notice.js

```diff
@@ -146,21 +166,47 @@ function writeLines(writeStream, lines) {
     }
 }
 
+/** Join zero or more iterables into a single iterable. */
+function* concat(...iterables) {
+    for (const it of iterables) {
+        yield* it;
+    }
+}
+
 function main(args) {
     try {
-        if (!(args && args.length > 2)) {
-            throw new Error(`Usage: node generate-third-party-notice.js <task name>`);
+        if (!args || args.length < 2) {
+            throw new Error(`Usage: node generate-third-party-notice.js <task name> [--verbose]`);
+        }
+
+        if (args.includes('--verbose')) {
+            verboseLogging = true;
         }
 
         const taskName = args[2];
         const taskPath = path.join(__dirname, 'Tasks', taskName);
         trace('task path', taskPath);
 
         const nodeModuleDir = path.join(taskPath, 'node_modules');
-        const licenseInfo = Array.from(collectLicenseInfo(nodeModuleDir));
+        const testsNodeModuleDir = path.join(taskPath, 'Tests', 'node_modules');
+        const licenseInfo = concat(
+            collectLicenseInfo(nodeModuleDir),
+            fs.existsSync(testsNodeModuleDir) ? collectLicenseInfo(testsNodeModuleDir) : []);
```

Comment:
Does this print out duplicate license information when a module is in both the task and test?  (If so, do we care?)

Reply:
Yes, it will produce duplicates.  I don't think it matters, and the list will be sorted so it will be obvious.  This happens in CondaEnvironmentV1 with the 'mockery' package (two different versions though).

I hope that including test dependencies is something we only have to do temporarily.  There's no need to distribute the test code, and if the build system can strip it out that will lighten our compliance burden.

Reply:
👍 FWIW, my guess is that we'd want the license for both versions (in case it changes).